eSims/eSims/Services/SubjectService.cs

```diff
@@ -8,20 +9,34 @@ namespace eSims.Services
 	public class SubjectService : ISubjectService
 	{
 		private readonly IMongoCollection<Subject> _subjects;
-		public SubjectService(IeSimsDatabaseSettings settings)
+
+        private readonly IMongoCollection<Professor> _professors;
+        public SubjectService(IeSimsDatabaseSettings settings)
 		{
 			var client = new MongoClient(settings.ConnectionString);
 			var database = client.GetDatabase(settings.DatabaseName);
 
 			_subjects = database.GetCollection<Subject>(settings.SubjectsCollectionName);
-		}
+            _professors = database.GetCollection<Professor>(settings.ProfessorsCollectionName);
+        }
 		public List<Subject> Get() =>
 			_subjects.Find(subject => true).ToList();
 		public Subject Get(string id) =>
 			_subjects.Find<Subject>(subject => subject.Id == id).FirstOrDefault();
 		public Subject Create(Subject subject)
 		{
-			_subjects.InsertOne(subject);
+            if (Get(subject.Id) != null)
+            {
+                return null;
+            }
```

Comment:
Don't do a Get operation inside a Create one. Do a simple function and extract the code from the Get operation and call that function in both Create and Get operation